app/services/InputCSVParserV3.java

```diff
@@ -0,0 +1,112 @@
+package services;
+
+import exceptions.GLLoadException;
+import play.Logger;
+import uk.co.onsdigital.discovery.model.DataPoint;
+import uk.co.onsdigital.discovery.model.DimensionValue;
+import uk.co.onsdigital.discovery.model.DimensionalDataSet;
+import uk.co.onsdigital.discovery.model.HierarchyEntry;
+
+import javax.persistence.EntityManager;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static services.InputCSVParser.END_OF_FILE;
+
+public class InputCSVParserV3 {
+
+    private static final Logger.ALogger logger = Logger.of(InputCSVParserV3.class);
+
+    public InputCSVParserV3() {
+    }
+
+    public void parseRowdataDirectToTablesFromTriplets(EntityManager em, String[] rowData, final DimensionalDataSet dds) {
+
+        String observation = getStringValue(rowData[0], "0");
+        if (END_OF_FILE.equals(observation)) {
+            logger.info("Found end-of-file marker");
+            return;
+        }
+
+        ArrayList<DimensionValue> dimensions = new ArrayList<>();
+
+        for (int i = 3; i < rowData.length; i = i + 3) {
+            String hierarchyId = rowData[i];
+            String dimensionName = rowData[i + 1];
+            String dimensionValue = rowData[i + 2];
+
+            logger.debug("Creating dimension for hierarchyId: " + hierarchyId + " and dimensionName: " + dimensionName + " and dimensionValue: " + dimensionValue + " ....");
```

Comment:
Should be:
logger.debug("Creating dimension for hierarchyId: {}  and dimensionName: {}and dimensionValue: {} ....",  hierarchyId, dimensionName, dimensionValue);